api/proto/scheduler/scheduler.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+enum RunnerType {
+  DOCKER = 0;
+}
+
+message Action {
+  string action_id = 1;
+  RunnerType runner_type = 2;
+  optional string image = 3;
+  repeated string commands = 4;
+}
+
+message RegisteredAgent {
+  string hostname = 1;
+  int32 cpu = 2;
+  int32 memory = 3;
+}
+
+message AgentId {
+  string id = 1;
+}
+
+message ActionLog {
+  string action_id = 1;
+  string log = 2;
+  optional bool success = 3; // Only populated if action is complete
+  optional int32 error_code = 4; // Only populated if success is false
+}
+
+message HealthStatus {
+  string agent_id = 1;
+  int32 cpu_usage = 2;
+  int32 memory_usage = 3;
+}
+
+message Empty {}
+
+service Scheduler {
+  rpc RegisterAgent (RegisteredAgent) returns (AgentId); // Agent
```

Comment:
```suggestion
  rpc RegisterAgent (RegisterAgentRequest) returns (RegisterAgentResponse); // Agent
```
I don't know if it is a best practise but i already saw it on many proto recently.

Reply:
changes done